### ISSUE TITLE
old ce loss issue come back

### ISSUE THREAD

**Issue by sipie800** [pytorch/pytorch#110264]
### 🐛 Describe the bug

https://github.com/pytorch/pytorch/issues/85005

This is happening at torch 2.x when training semantic segmentation with large input like shape(8, 171, 512, 512).
regular input shape won't trigger it.

I can make it happens very few times by implementing ce calculating by reducing pieces of input. But it just won't disappear unless the input shape is very small like a classification task.

And inserting some trash operation like print(a large tensor) before loss calculating will make it disappear, which just makes training way too slow

### Versions

torch 2.0.1 win10 cuda118 rtx4090

And I found it's also in torch 1.13.1 cuda117 too
